interface/modules/custom_modules/oe-module-claimrev-connect/ModuleManagerListener.php

```diff
@@ -0,0 +1,232 @@
+    public static function initListenerSelf(): ModuleManagerListener
+    {
+        return new self();
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function install($modId, $currentActionStatus): mixed
+//    {
+//        $modService = new ModuleServices();
+//        /* setting the active ui flag here will allow the config button to show
+//         * before enable. This is a good thing because it allows the user to
+//         * configure the module before enabling it. However, if the module is disabled
+//         * this flag is reset by MM.
+//        */
+//        $modService::setModuleState($modId, '0', '1');
+//        return $currentActionStatus;
+//    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+    private function help_requested($modId, $currentActionStatus): mixed
+    {
+        // must call a script that implements a dialog to show help.
+        // I can't find a way to override the Lamina's UI except using a dialog.
+        if (file_exists(__DIR__ . '/show_help.php')) {
+            include __DIR__ . '/show_help.php';
+        }
+        return $currentActionStatus;
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function preenable($modId, $currentActionStatus): mixed
+//    {
+//        return $currentActionStatus;
+//    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function enable($modId, $currentActionStatus): mixed
+//    {
+//        $modService = new ModuleServices();
+//        if ($modService->isClaimRevConfigured()) {
+//            $modService::setModuleState($modId, '1', '0');
+//            return $currentActionStatus;
+//        }
+//        $modService::setModuleState($modId, '1', '1');
+//        return xlt("Weno eRx Service is not configured. Please configure Weno eRx Service in the Weno Module Setup.");
+//    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function disable($modId, $currentActionStatus): mixed
+//    {
+//        // allow config button to show before enable.
+//        ModuleServices::setModuleState($modId, '0', '1');
+//        return $currentActionStatus;
+//    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+    private function unregister($modId, $currentActionStatus): mixed
+    {
+        $sql = "DELETE FROM `background_services` WHERE `name` = ? OR `name` = ? OR `name` = ?";
+        sqlQuery($sql, array('ClaimRev_Send', 'ClaimRev_Receive', 'ClaimRev_Elig_Send_Receive'));
+        return $currentActionStatus;
+    }
+
+    /**
+     * @param $modId
+     * @param $currentActionStatus
+     * @return mixed
+     */
+//    private function reset_module($modId, $currentActionStatus): mixed
```

Comment:
If needed or wanted, this gets called also when unregistered action is called.

Reply:
When I enabled all the methods that were commented out. The module will not install. It can enable but it won't install. I have uploaded what I have. The code does not produce any errors in the errors in the logs.

Reply:
I will try to figure this out tomorrow. I got some other stuff to work on right now.

Reply:
@juggernautsei Just delete the unused methods(commented). The controller will simply ignore them. You need unregister, enable and disable for task maintenance.

Reply:
I figured out what you were saying and just return current status. I have installed and removed. It removes all the globals and tables.

Reply:
I wanted a cleanup action that I chose to name reset i.e. reset module to a state before original install. All these still need to be documented but besides our admins, you and Brad, I've seen no third party user make any attempt to give feedback or contrib.
So, you're in the know now so to say:)

Reply:
@sjpadgett I move the getModuleState method into the listener. The reset_module method is not being called. I checked the error log and it does not show there. The background services are being removed but not the table or globals as you would know.

Reply:
Forget reset_module just delete it you won't be using.
You don't need getModuleState. I wrote those for how I use weno module actions.
All you need is enable method with the sql to set the 3 active flag in background services, disable to unset active flag in background service table and what you already have to delete the tasks.

Reply:
got it and here are the updates.